aioseop_class.php

```diff
@@ -1471,11 +1471,17 @@ function get_aioseop_title( $post, $use_original_title_format = true ) {
 				$title = $this->internationalize( get_option( 'blogname' ) ) . ' | ' . $this->internationalize( get_bloginfo( 'description' ) );
 			}
 
+			// #1616 - Avoid trying to get property of non-object when no posts are present on the homepage
 			global $post;
-			$post_id = $post->ID;
+			$frontpage_id = get_option( 'page_on_front' );
+
+			if ( $post === null ) {
+				$post_id = $frontpage_id;
```

Comment:
You could probably remove `$frontpage_id` all together by replacing this with.

```
$post_id = get_option( 'page_on_front' );
```

Reply:
I've made this change